storage/replica.go

```diff
@@ -1028,6 +1028,11 @@ func (r *Replica) endCmds(cmd *cmd, ba *roachpb.BatchRequest, br *roachpb.BatchR
 					// DeleteRange adds to the write timestamp cache to prevent
 					// subsequent writes from rewriting history.
 					readTSCache = false
+					// DeleteRange requires that we retry on push to avoid the
+					// lost delete range anomaly.
+					if br.Txn != nil {
+						br.Txn.RetryOnPush = true
```

Comment:
Can you move this into a txn returned by `DeleteRange` itself? This is an odd place.